### PR TITLE
Rename `configure_azure_monitor`, add sampler and export interval to config

### PR DESCRIPTION
**Pull request by lzchen**

### CODE REVIEW

**Review by JacksonWeber — APPROVED** [review]

File: azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/__init__.py [hunk 1]
```diff
@@ -4,29 +4,33 @@
...
     This function works as a configuration layer that allows the
     end user to configure OpenTelemetry and Azure monitor components. The
-    configuration can be done via environment variables or
-    via arguments passed to this function. Each argument has a 1:1
-    correspondence with an environment variable.
+    configuration can be done via arguments passed to this function.
     """
 
     configurations = get_configurations(**kwargs)
-    connection_string = configurations["connection_string"]
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/__init__.py:25]: Why remove conn str?

File: azure-monitor-opentelemetry-distro/samples/tracing/simple.py [hunk 2]
```diff
@@ -4,10 +4,14 @@
...
 
-from azure.monitor.opentelemetry.distro import configure_opentelemetry
+from azure.monitor.opentelemetry.distro import configure_azure_monitor
 from opentelemetry import trace
 
-configure_opentelemetry()
+configure_azure_monitor(
+    connection_string="<your-connection-string>",
+    service_name="foo_service",
+    tracing_export_interval=15,
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry-distro/samples/tracing/simple.py:13]: I believe this is in milliseconds. Did you mean 15000?

Comment (jeremydvoss) [reply azure-monitor-opentelemetry-distro/samples/tracing/simple.py:13]: We may also want to consider including the scale in the name, but if not, documentation will cover it.

File: azure-monitor-opentelemetry-distro/tests/configuration/test_configure.py [hunk 3]
```diff
@@ -50,14 +54,18 @@ def test_configure_opentelemetry(
         exporter_mock.return_value = exp_init_mock
         resource_init_mock = Mock()
         resource_mock.create.return_value = resource_init_mock
+        sampler_init_mock = Mock()
+        sampler_mock.return_value = sampler_init_mock
         bsp_init_mock = Mock()
         bsp_mock.return_value = bsp_init_mock
-        configure_opentelemetry(
+        configure_azure_monitor(
             connection_string="test_cs",
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry-distro/tests/configuration/test_configure.py:62]: I think this conn str no longer works, right?

File: azure-monitor-opentelemetry-distro/tests/configuration/test_configure.py [hunk 4]
```diff
@@ -50,14 +54,18 @@ def test_configure_opentelemetry(
...
             disable_tracing=False,
             service_name="test_service_name",
             service_namespace="test_namespace",
             service_instance_id="test_id",
+            sampling_ratio=0.5,
+            tracing_export_interval=15,
```
Comment (jeremydvoss) [inline azure-monitor-opentelemetry-distro/tests/configuration/test_configure.py:68]: Milliseconds. It's just a value for testing, but figured you may want to change it here, too

File: azure-monitor-opentelemetry-distro/tests/configuration/test_configure.py (hunk 3 above)
Comment (lzchen) [reply azure-monitor-opentelemetry-distro/tests/configuration/test_configure.py:62]: Moved this to the exporter test.

File: azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/__init__.py (hunk 1 above)
Comment (lzchen) [reply azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/__init__.py:25]: We are taking advantage of passing in the whole `kwargs` into the exporter so no need to explicitly extract the connection string.

**Review by jeremydvoss — APPROVED** [review]

**Pull request merged**